process/database.go

```diff
@@ -106,32 +106,33 @@ func CleanProcessStorage(activePIDs map[int]struct{}) {
 
 	// clean primary processes
 	for _, p := range processesCopy {
-		p.Lock()
+		// The PID of a process does not change.
 
+		// Check if this is a special process.
+		if p.Pid == UnidentifiedProcessID || p.Pid == SystemProcessID {
+			p.profile.MarkStillActive()
+			continue
+		}
+
+		// Check if process is active.
 		_, active := activePIDs[p.Pid]
+		if active {
+			p.profile.MarkStillActive()
+			continue
+		}
+
+		// Process is inactive, start deletion process
 		switch {
-		case p.Pid == UnidentifiedProcessID:
-			// internal
-		case p.Pid == SystemProcessID:
-			// internal
-		case active:
-			// process in system process table or recently seen on the network
+		case p.LastSeen == 0:
+			// add last
+			p.LastSeen = time.Now().Unix()
+		case p.LastSeen > threshold:
+			// within keep period
```

Comment:
Event though the PID is immutable we are mutating LastSeen so I guess the profile should be locked?